pennylane/ops/qubit/observables.py

```diff
@@ -488,10 +565,190 @@ def compute_diagonalizing_gates(
 
         **Example**
 
-        >>> qml.Projector.compute_diagonalizing_gates([0, 1, 0, 0], wires=[0, 1])
+        >>> _BasisStateProjector.compute_diagonalizing_gates([0, 1, 0, 0], wires=[0, 1])
         []
         """
         return []
 
-    def pow(self, z):
-        return [copy(self)] if (isinstance(z, int) and z > 0) else super().pow(z)
+
+class _StateVectorProjector(Observable):
+    r"""
+    Observable corresponding to the computational basis state projector :math:`P=\ket{\phi}\bra{\phi}`.
+
+    The expectation of this observable returns the value
+
+    .. math::
+        |\langle \psi | \phi \rangle |^2
+
+    corresponding to the probability of measuring the quantum state in the basis defined by :math:`\phi`.
+
+    For example, the projector :math:`\ket{+}\bra{+}` is created by ``state_vector=np.array([1, 1])/np.sqrt(2)``.
+
+    **Details:**
+
+    * Number of wires: Any
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        state_vector (tensor-like): state vector of shape ``(n, )``.
+        wires (Iterable): wires that the projector acts on.
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional).
+        id (str or None): String representing the operation (optional).
+    """
```

Comment:
Note that private underscored classes don't strictly require docstrings. I'd actually remove the current docstring here and only document anything specifically that we'd want a developer to read (if anything).

As it is now, it's just more to maintain (e.g., if we make a change to the docstring do we have to make in 3 places?)

Reply:
I makes a lot of sense! Done :)